casepro/statistics/models.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import unicode_literals
+
+import pytz
+
+from dash.orgs.models import Org
+from django.contrib.auth.models import User
+from django.db import models, connection
+from django.db.models import Sum
+from django.utils.translation import ugettext_lazy as _
+
+from casepro.cases.models import Partner
+
+
+class BaseDailyCount(models.Model):
+    """
+    Base class for models which record counts of something per day
+    """
+    TYPE_REPLIES = 'R'
+
+    type = models.CharField(max_length=1)
+
+    day = models.DateField(help_text=_("The day this count is for"))
+
+    count = models.PositiveIntegerField(default=1)
+
+    squashed = models.BooleanField(default=False)
+
+    @classmethod
+    def squash(cls):
+        """
+        Squashes counts so that there is a single count per unique field combination per day
+        """
+        unique_fields = list(cls.UNIQUE_FIELDS) + ['type', 'day']
+        unsquashed_values = cls.objects.filter(squashed=False).values(*unique_fields).distinct(*unique_fields)
+
+        for unsquashed in unsquashed_values:
+            with connection.cursor() as cursor:
```

Comment:
I appreciate these aren't the most readable dozen or so lines, but couldn't find another way of doing this that was both atomic, and generic, i.e. don't have duplicate for every different count model.